### PR TITLE
fix: single unread event in conversation last message [AR-3003]

### PR DESCRIPTION
**Pull request by Garzas**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-3003" title="AR-3003" target="_blank">StoryAR-3003</a>  Add an icon for missed events in the conversation list item
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

When there was only one type of unread event with counter equal to one there was message for example `1 mention` instead of `Someone mentioned you`. So I fixed and simplified logic to be more understandable. 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1563/1/) **failed**.

File: kalium
Comment (vitorhugods) [inline kalium:1]: ⚠️ Kalium reference pointed to a branch different than `develop`

**Review by vitorhugods — CHANGES_REQUESTED** [review]
👨‍🍳👌

Just check the Kalium reference

File: app/src/main/kotlin/com/wire/android/mapper/MessagePreviewContentMapper.kt [hunk 1]
```diff
@@ -38,61 +38,61 @@ fun MessagePreview?.toUIPreview(unreadEventCount: UnreadEventCount): UILastMessa
...
+                UnreadEventType.IGNORED -> null
+                null -> null
             }
-        }
-    }
+        }.associate { it }
 
-    return uiLastMessageContent()
+    val first = unreadContentTexts.values.first()
+    val second = unreadContentTexts.values.elementAt(1)
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/mapper/MessagePreviewContentMapper.kt]: I believe this method will be called also when `unreadEventCount.size == 1 && unreadEventCount.keys.first() != UnreadEventType.MESSAGE && unreadEventCount.values.first() != 1`, in that case there won't be any element at position 1

**Review by gongracr — APPROVED** [review]
LGTM 🥒 🍅

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1563/2/) **failed**.

File: app/src/main/kotlin/com/wire/android/mapper/MessagePreviewContentMapper.kt (hunk 1 above)
Comment (Garzas) [reply app/src/main/kotlin/com/wire/android/mapper/MessagePreviewContentMapper.kt]: Good catch 👨‍🏫

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1563/3/) **failed**.

**Review by saleniuk — APPROVED** [review]

**Review by vitorhugods — APPROVED** [review]
👨‍🍳👌

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1563/4/) **failed**.

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1563/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.2(6042816)(6042816)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.2(6042816)(6042816)-staging-compat.apk)

**Pull request merged**